pkg/util/leaderlease/leaderlease.go

```diff
@@ -63,8 +65,29 @@ func NewEtcd(client etcdclient.Client, key, value string, ttl uint64) Leaser {
 	}
 }
 
+const autoSyncInterval = 10 * time.Second
+
 // AcquireAndHold implements an acquire and release of a lease.
 func (e *Etcd) AcquireAndHold(notify chan error) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	go func() {
+		// Because the call to e.keysClient.Set in tryAcquire is using PrevNoExist, etcd considers this
+		// to be a "one-shot" attempt, meaning that if the connection attempt to one of the etcd cluster
+		// members fails, it will not fail over to any of the other cluster members. Calling
+		// e.client.AutoSync is not a one-shot call, and it will try to contact each cluster member
+		// until it succeeds. Assuming it does, the client's list of endpoints is updated, and any
+		// unavailable members are removed from the list.
+		for {
+			err := e.client.AutoSync(ctx, autoSyncInterval)
```

Comment:
You're right - if the inner call to Sync fails repeatedly, it would be a hotloop. At the very least, after an error, it should sleep for e.pauseInterval.